hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultPortableReader.java

```diff
@@ -684,8 +687,11 @@ private PortablePosition findPositionForReading(String path) throws IOException
         }
     }
 
-    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:returncount"})
+    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:returncount", "unchecked"})
     private <T> T readSinglePositionFromNonArray(PortablePosition position) throws IOException {
+        if (position.getType() == null) {
+            throw new IllegalArgumentException("Unsupported type read: null");
+        }
         switch (position.getType()) {
```

Comment:
I already use them everywhere and had a discussion about them. There isn't a definite policy, so it's up to you. Although it may not seem like much difference, all the checks can eventually add up.